processor/src/trace/decoder/mod.rs

```diff
@@ -227,14 +234,15 @@ fn build_aux_col_p3<E: FieldElement<BaseField = Felt>>(
     trace_len: usize,
     aux_trace_hints: &AuxTraceHints,
     alphas: &[E],
+    main_trace: &Matrix<Felt>,
```

Comment:
I think `main_trace` should be the first parameter to this function like it is for `build_aux_col_p1` and `build_aux_col_p2`